evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py

```diff
@@ -0,0 +1,466 @@
+        assert all(isinstance(x, pd.DataFrame) for x in result_dfs)
+        assert all(get_trend_dataframe_format_correct(x) for x in result_dfs)
+        if variateness == "univariate":
+            assert len(result_dfs) == 1
+            [get_trend_dataframe_format_correct(x) for x in result_dfs]
+        elif variateness == "multivariate":
+            assert len(result_dfs) == 2
+            [get_trend_dataframe_format_correct(x) for idx, x in enumerate(result_dfs)]
+
+    elif transformer_fit_on_data != "in-sample":
+
+        y_t_new = build_test_target(
+            subset_y,
+            seasonal_period,
+            transformer_fit_on_data,
+            to_test="transform",
+        )
+        dec = STLDecomposer()
+        dec.fit(subset_X, subset_y)
+
+        # get_trend_dataframe() is only expected to work with datetime indices
+        if variateness == "multivariate":
+            y_t_new = pd.concat([y_t_new, y_t_new], axis=1)
+
+        if transformer_fit_on_data in [
+            "out-of-sample-in-past",
+            "partially-out-of-sample-in-past",
+        ]:
+            with pytest.raises(
+                ValueError,
+                match="STLDecomposer cannot transform/inverse transform data out of sample",
+            ):
+                result_dfs = dec.get_trend_dataframe(X, y_t_new)
+
+        else:
+            result_dfs = dec.get_trend_dataframe(X.loc[y_t_new.index], y_t_new)
+
+            assert isinstance(result_dfs, list)
+            assert all(isinstance(x, pd.DataFrame) for x in result_dfs)
+            assert all(get_trend_dataframe_format_correct(x) for x in result_dfs)
+            if variateness == "univariate":
+                assert len(result_dfs) == 1
+                [get_trend_dataframe_format_correct(x) for x in result_dfs]
+            elif variateness == "multivariate":
+                assert len(result_dfs) == 2
+                [
+                    get_trend_dataframe_format_correct(x)
+                    for idx, x in enumerate(result_dfs)
+                ]
+
+
+def test_stl_decomposer_get_trend_dataframe_sets_time_index_internally(
+    generate_seasonal_data,
+):
+    def get_trend_dataframe_format_correct(df):
+        return set(df.columns) == {"signal", "trend", "seasonality", "residual"}
+
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+        period=7,
+        set_time_index=False,
+    )
+    assert not isinstance(y.index, pd.DatetimeIndex)
+
+    stl = STLDecomposer()
+    stl.fit(X, y)
+    result_dfs = stl.get_trend_dataframe(X, y)
+
+    assert isinstance(result_dfs, list)
+    assert all(isinstance(x, pd.DataFrame) for x in result_dfs)
+    assert all(get_trend_dataframe_format_correct(x) for x in result_dfs)
+
+
+@pytest.mark.parametrize(
+    "bad_frequency",
+    [
+        pytest.param(
+            "T",
+            marks=pytest.mark.xfail(
+                reason="statsmodels freq_to_period doesn't support this frequency",
+            ),
```

Comment:
This is such an interesting way to test this failure. Should we add code within the STLDecomposer to explicitly throw an error if it receives these frequencies instead, to make it more explicit?

Reply:
I think I'm going to have to forego this one for now, pending how we actually want to handle the various `freq_to_period` functions in pandas and statsmodels